release: 0.6.0 (#2662)
rome/tools: CHANGELOG.md

# Rome changelog

## 0.6.0

- BREAKING CHANGES: the command `rome format --ci` has been removed, use `rome ci` instead.
- Improved the compatibility with Prettier (check #2403 for more details):

>   - TypeScript's formatting is now in line with what Prettier does.

Author: I just relied on #2403, where we addressed all TS bugs. Should we reword this bullet point?

>   - TypeScript's formatting is now in line with what Prettier does.

Author: I reworded the bullet point

>   - Better formatting of string literals. The formatter now does a lot of cleaning:

Reviewer: Removing unnecessary quotes in string literals and quotes from member names.

Author: I reworded the bullet point and added another example

  ```js
  // original code
  let a = {
    "something": 3
  }

>   let b = "cool ins\'t it";

Reviewer: ```suggestion
  let b = "cool isn\'t it";
```

  
  // formatted code
  let a = {

>     someething: 3

Reviewer 2: Typo:
```suggestion
    something: 3
```

  }
  let b = "cool ins't it";
  ```
  - Better formatting of various statements
  - Improved the performance of the formatting of averagely 20%! Check the [relative

> PR](https://github.com/rome/tools/pull/2634) if you're interested of what the team did.

Author: Feel free to add a suggestion and I will merge it

> To reach a better compatibility with Prettier, the team had to review the foundation of our printer,

Author: Sure, feel free to submit a suggestion

> To reach a better compatibility with Prettier, the team had to review the foundation of our printer,

Reviewer 3: ```suggestion
To reach better compatibility with Prettier, the team had to revise the foundation of our printer,
```

# Rome changelog

## 0.6.0

### Formatter

- BREAKING CHANGES: the command `rome format --ci` has been removed, use `rome ci` instead.

> - Improved the compatibility with Prettier (check #2403 for more details):

Reviewer: Should we make this the headline under the formatter title. Because that's the focus of this release. 

It would allow you then to remove the sub-bullet points. 

Nit: I personally wouldn't mention typescript explicitly. Saying that we got closer to prettier summarises this already

  formatter/math.js                        1.27    252.5±2.95ms     2.6 MB/sec    1.00    198.7±2.22ms     3.3 MB/sec
  formatter/parser.ts                      1.33      5.7±0.02ms     8.5 MB/sec    1.00      4.3±0.02ms    11.2 MB/sec
  formatter/pixi.min.js                    1.38    140.2±1.68ms     3.1 MB/sec    1.00    101.7±2.14ms     4.3 MB/sec
  formatter/react-dom.production.min.js    1.33     39.1±0.62ms     2.9 MB/sec    1.00     29.4±0.42ms     3.9 MB/sec
  formatter/react.production.min.js        1.34   1931.6±3.99µs     3.2 MB/sec    1.00  1444.7±28.41µs     4.3 MB/sec
  formatter/router.ts                      1.26      4.2±0.01ms    14.5 MB/sec    1.00      3.3±0.02ms    18.4 MB/sec
  formatter/tex-chtml-full.js              1.20    313.5±1.08ms     2.9 MB/sec    1.00    260.8±2.10ms     3.5 MB/sec
  formatter/three.min.js                   1.36    163.0±1.68ms     3.6 MB/sec    1.00    119.7±2.08ms     4.9 MB/sec
  formatter/typescript.js                  1.34   1033.0±2.53ms     9.2 MB/sec    1.00    772.2±6.47ms    12.3 MB/sec
  formatter/vue.global.prod.js             1.34     50.8±0.51ms     2.4 MB/sec    1.00     37.9±1.11ms     3.2 MB/sec

>   ```

Reviewer: I would remove that block as it may be misleading. This measures pure formatting time which isn't what users may experience when using the CLI where IO is a significant factor too.
